tablo/models.py

```diff
@@ -156,26 +182,25 @@ def _get_time_extent(self):
 
     @property
     def fields(self):
-        fields = []
-        with connection.cursor() as c:
-            c.execute('select column_name, is_nullable, data_type from information_schema.columns where table_name = %s;', [self.table])
-            # c.description won't be populated without first running the query above
-            for field_info in c.fetchall():
-                field_type = field_info[2]
-                if field_info[0] == 'db_id':
-                    field_type = 'OID'
-                elif field_info[0] == POINT_FIELD_NAME:
-                    field_type = 'Geometry'
-
-                fields.append({
-                    'name': field_info[0],
-                    'alias': field_info[0],
-                    'type': POSTGIS_ESRI_FIELD_MAPPING[field_type],
-                    'nullable': True if field_info[1] == 'YES' else False,
-                    'editable': True
-                })
-
-        return fields
+        if self._fields is None:
+            fields = get_fields(self.table)
+            for field in fields:
+                if field['name'] == 'db_id':
+                    field['type'] = 'esriFieldTypeOID'
+                elif field['name'] == POINT_FIELD_NAME:
+                    field['type'] = 'esriFieldTypeGeometry'
+            self._fields = fields
+
+        return self._fields
+
+    @property
+    def related_fields(self):
+        if self._related_fields is None:
+            self._related_fields = {
```

Comment:
This hurts my brain. It's nice and compact but as a reader of this code I'm going to have to stop and think on this one every time.

The alternative:
```
self._related_fields = {}
for relationship in self.featureservicelayerrelations_set.all():
    for field in relationship:
        key = '{table}.{field}'.format(table=relationship.related_title, field=field['name'])
        self.related_fields[key] = field
```

seems more readable to me.